From: Duncan Eastoe <duncan@graphiant.com>
Date: Tue, 2 Aug 2022 14:08:00 +0100
Subject: common: add SR_VAL_LY2SR_UNDERLYING_VALUE option

Add an option for use with sr_val_ly2sr_with_opts() which converts
nodes of type LY_TYPE_ENUM to SR_INT32_T, rather than SR_ENUM_T.
---
 src/common.c | 11 ++++++++---
 src/common.h |  3 +++
 2 files changed, 11 insertions(+), 3 deletions(-)

diff --git a/src/common.c b/src/common.c
index 19037bc..cffe1e7 100644
--- a/src/common.c
+++ b/src/common.c
@@ -4032,9 +4032,14 @@ store_value:
             sr_val->data.string_val = NULL;
             break;
         case LY_TYPE_ENUM:
-            sr_val->type = SR_ENUM_T;
-            sr_val->data.enum_val = strdup(lyd_value_get_canonical(LYD_CTX(node), val));
-            SR_CHECK_MEM_GOTO(!sr_val->data.enum_val, err_info, error);
+            if (opts & SR_VAL_LY2SR_UNDERLYING_VALUE) {
+                sr_val->data.int32_val = val->int32;
+                sr_val->type = SR_INT32_T;
+            } else {
+                sr_val->type = SR_ENUM_T;
+                sr_val->data.enum_val = strdup(lyd_value_get_canonical(LYD_CTX(node), val));
+                SR_CHECK_MEM_GOTO(!sr_val->data.enum_val, err_info, error);
+            }
             break;
         case LY_TYPE_IDENT:
             sr_val->type = SR_IDENTITYREF_T;
diff --git a/src/common.h b/src/common.h
index f4c7420..3a1e39c 100644
--- a/src/common.h
+++ b/src/common.h
@@ -1374,6 +1374,9 @@ sr_error_info_t *sr_conn_push_oper_mod_del(sr_conn_ctx_t *conn, const char *mod_
 
 typedef enum sr_val_ly2sr_opts_e {
     SR_VAL_LY2SR_DEFAULT = 0,             /**< Standard behaviour */
+    SR_VAL_LY2SR_UNDERLYING_VALUE = 1,    /**< Convert nodes of type LY_TYPE_ENUM to SR_INT32_T, with the value being
+                                               the underlying numeric representation of the enum value, rather than
+                                               SR_ENUM_T. There is currently no effect on any other node type. */
 } sr_val_ly2sr_opts_t;
 
 /**
